### ISSUE TITLE
Side effect of the set method inside the PVector class

### ISSUE THREAD

**Issue by Maharal** [processing/processing#5092]
If the user passes a vector with a size different than 2 and 3 as a parameter to the _set_ method, there will be **side effects**.

```java
// line 197, https://github.com/processing/processing/blob/master/core/src/processing/core/PVector.java
public PVector set(float[] source) {
  if (source.length >= 2) {
    x = source[0];
    y = source[1]; 
  }
  if (source.length >= 3) {
    z = source[2];
  }
  return this;
}
```

**Comment by GKFX** [comment]
What side effects? It looks like the later members of source just get ignored.

**Comment by benfry** [comment]
It's just an esoteric edge case where if someone had a 3D vector and decided to set it using a 2D vector, the z coordinate wouldn't get zeroed out. I've added a fix for 3.3.4.

**Comment by GoToLoop** [comment]
Zeroing the field _z_ for **set(float, float)** & **set(float[])** w/ _length_ = 2 @ version 3.3.4, breaks any older code, such as Pjs, which would expect _z_'s current value to be kept for those overloaded cases.
